src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -312,6 +328,13 @@ def add_title(self, title: str) -> VideoSegmentResource:
         self.title = title
         return self
 
+    def add_title_optional(self, title: Any) -> VideoSegmentResource:
+        if not pd.isna(title):
+            if self.title:
+                _warn_value_exists(self.title, title, "title", self.res_id)
```

Comment:
Calling a function with 4 different parameters of the same type, without using keywords, is prone to errors. Nothing will warn you if you mistakenly flip the order of the parameters.

Suggestion: make these parameters keyword-only. You can do this by prepending a `*` to the parameter list:
```
def _warn_value_exists(*, old_value: Any, new_value: Any, value_field: str, res_id: str | None) -> None
```

You will then be forced to always use keyword parameters